apps/prairielearn/src/sync/fromDisk/courseInstances.ts

```diff
@@ -38,6 +41,41 @@ export async function sync(
   courseId: string,
   courseData: CourseData,
 ): Promise<Record<string, string>> {
+  if (config.checkInstitutionsOnSync) {
+    // Collect all institutions from course instance access rules.
+    const institutions = Object.values(courseData.courseInstances)
+      .flatMap(({ courseInstance }) => courseInstance.data?.allowAccess)
+      .map((accessRule) => accessRule?.institution)
+      .filter((institution) => institution != null);
+
+    // Select only the valid institution names.
+    const validInstitutions = await sqldb.queryRows(
+      sql.select_valid_institution_short_names,
+      { short_names: Array.from(new Set(institutions)) },
+      z.string(),
+    );
+
+    const validInstitutionSet = new Set(validInstitutions);
+
+    // This is a special hardcoded value that is always valid.
+    validInstitutionSet.add('Any');
+
+    // Add sync warnings for invalid institutions.
+    Object.values(courseData.courseInstances).forEach(({ courseInstance }) => {
+      const instanceInstitutions = new Set(
+        courseInstance.data?.allowAccess
+          ?.map((accessRule) => accessRule?.institution)
+          .filter((institution) => institution != null),
+      );
+
+      instanceInstitutions.forEach((institution) => {
+        if (validInstitutionSet.has(institution)) return;
+
+        infofile.addWarning(courseInstance, `Institution "${institution}" not found.`);
```

Comment:
I'm treating these as warnings, not errors; does that make sense?

Reply:
I'm inclined to say that it should be an error. I can't think of any reason that you'd want to have an invalid institution listed? It's almost certainly not going to do what the instructor intended, which to me says "error"?

Reply:
Changed to an error in 6833160.